Project setup

sta-wtag opened the pull request:
nuxt project setup , i18n setup, store setup, vue test utils setup, eslint setup, add task feature

sta-wtag committed:
Setup: project setup complete, sass added

sta-wtag committed:
Setup:vue setup , i18n setup complete

sta-wtag committed:
Setup: vue test utils setup cone using vue cli

sta-wtag committed:
i18n:multilingual setup added

sta-wtag committed:
add: add task done

sta-wtag committed:
feature: search task

sta-wtag committed:
add: add task completed

hhu-wtag commented on .eslintrc.js:
```diff
@@ -0,0 +1,17 @@
+        parser: 'babel-eslint'
+    },
+    extends: [
+        //'@nuxtjs',
+        //'plugin:nuxt/recommended',
+        //'plugin:vue/recommended',
```
We can remove the comment if not needed.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,49 @@
+          </svg>
+        </div>
+        <div>
+          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
+            <path d="M7.127 22.562L0 24L1.438 16.872L7.127 22.562ZM8.541 21.148L19.769 9.923L14.079 4.231L2.852 15.458L8.541 21.148ZM18.309 0L15.493 2.817L21.184 8.508L24 5.689L18.309 0V0Z" fill="#BBBDD0"/>
+          </svg>
```
We can create separate component for the SVG icons and use them so that our markup is minimal and friendly to first look.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,49 @@
+          </svg>
+        </div>
+        <div>
+          <svg xmlns="http://www.w3.org/2000/svg" width="20" height="24" viewBox="0 0 20 24" fill="none">
+            <path fill-rule="evenodd" clip-rule="evenodd" d="M17 24H3C1.896 24 1 23.104 1 22V6H19V22C19 23.104 18.104 24 17 24ZM8 10C8 9.448 7.552 9 7 9C6.448 9 6 9.448 6 10V19C6 19.552 6.448 20 7 20C7.552 20 8 19.552 8 19V10ZM14 10C14 9.448 13.552 9 13 9C12.448 9 12 9.448 12 10V19C12 19.552 12.448 20 13 20C13.552 20 14 19.552 14 19V10ZM20 5H0V3H6V1.5C6 0.673 6.673 0 7.5 0H12.5C13.325 0 14 0.671  ...
+          </svg>
```
ditto

sta-wtag replied:
Resolved

hhu-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,49 @@
+    name:'TaskCard',
+    computed: {
+      singleTask(){
+        return this.$store.state.todo.states.task
```
We can use [mapGetters](https://github.com/ana-wtag/todo-app-vue/pull/1#discussion_r1083655280) here

sta-wtag replied:
Resolved

hhu-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,49 @@
+        return this.$store.state.todo.states.task
+      }
+    },
+    props:{
+      cardData:{
+        type:Object,
+        default:null
+      }
+    },
+    data:()=> ({
```
Something is wrong with code alignment here. I think prettier is acting up from your side. Please check.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+<template>
+    <div class="header">
+        <div >
+            <svg xmlns="http://www.w3.org/2000/svg" width="34" height="34" viewBox="0 0 34 34" fill="none">
+                <g clip-path="url(#clip0_8_429)">
+                <path d="M33.5793 4.4657C29.9019 1.67903 24.2431 0.0153503 18.4416 0.0153503C11.2658 0.0153503 5.4185 2.52072 2.39921 6.88906C0.981168 8.94068 0.196844 11.3701 0.0680823 14.1099C-0.0466013 16.5494 0.363059 19.2473 1.28611 22.1499C4.43622 12.7074 13.2333 5.31251 23.375 5.31251C23.375 5.31251 13.8847 7.81045 7.91816 15.5454C7.91444 15.55 7.83535 15.648 7.69856 15.8342C6.50059 17.437 ...
+                </g>
+                <defs>
+                <clipPath id="clip0_8_429">
+                <rect width="34" height="34" fill="white"/>
+                </clipPath>
+                </defs>
+            </svg>
```
Ditto

sta-wtag replied:
Resolved

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+        </div>
+        <div style="display:flex;">
+            <div @click="startAnim" id="searchIcon">
+                <svg  xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
+                    <path d="M23.111 20.058L18.134 15.081C19.099 13.561 19.657 11.759 19.657 9.83C19.657 4.41 15.248 0 9.828 0C4.408 0 0 4.41 0 9.83C0 15.25 4.408 19.66 9.829 19.66C11.663 19.66 13.381 19.155 14.851 18.277L19.872 23.298C22.016 25.439 25.256 22.202 23.111 20.058ZM3.047 9.83C3.047 6.091 6.09 3.048 9.829 3.048C13.568 3.048 16.611 6.09 16.611 9.83C16.611 13.57 13.568 16.612 9.829 16.6 ...
+                </svg>
```
Ditto

sta-wtag replied:
Resolved

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+            </div>
+        <select class="" @change="localePathAbout">
+            <option v-for="locale in $i18n.locales" :key="locale.code">
+                <!-- <nuxt-link :to="localePathAbout(locale.code)"> -->
```
Remove if not needed

sta-wtag replied:
Resolved

bia-wtag commented on .eslintrc.js:
```diff
@@ -0,0 +1,17 @@
+        'plugin:vue/essential',
+        'eslint:recommended'
+    ],
+}
```
newline expected.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+    },
+    startAnim() {
+        this.showInput = true;
+        // document.getElementById('searchIcon').className='searchIcon'
```
Remove if not needed

sta-wtag replied:
Resolved

bia-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,49 @@
+    <div class="card">
+      <div class="cardDescription">{{ cardData.description }}</div>
+      <div style="margin-top:9px; margin-bottom:26px;" class="caption">{{ cardData.createdAt }}</div>
+      <div style="display:flex; gap:8px;">
```
Don't use a style like this. Add these properties in `<style>` section or create classes.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+        searchInput: {
+            handler(val)
+                {
+                 this.setTime= setTimeout(() => { 
```
Why are we using setTimeout here? For debouncing? Or to mimic API response?

sta-wtag replied:
mimic API response

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+                    this.$store.commit('todo/setFilteredList',test)
+                    $nuxt.$emit('searchOn')
+                   }, 1000)  
+                   //clearTimeout(this.setTime);
```
ditto

sta-wtag replied:
Resolved

bia-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,49 @@
+</script>
+<style lang="scss">
+   
+</style>
```
We can remove this if empty.

sta-wtag replied:
resolved

bia-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,49 @@
+        default:null
+      }
+    },
+    data:()=> ({
```
Fix formatting.

sta-wtag replied:
Resolved

hhu-wtag commented on tests/unit/TaskCard.spec.js:
```diff
@@ -0,0 +1,24 @@
+import Nuxti18n from '@nuxtjs/i18n'
+
+
+// const i18n = new Nuxti18n({
+//   locale: 'en',
+// })
```
Remove if not needed

sta-wtag replied:
Resolved

bia-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+    methods: {
+    localePathAbout(event) {
+       if(event.target.value === 'en')
+      {return  this.$router.replace(this.switchLocalePath('en'))}
```
Can we skip route paths when translating?

sta-wtag replied:
yes. Resolved

bia-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+        this.showInput = true;
+        // document.getElementById('searchIcon').className='searchIcon'
+    },
+    handleChange() {
```
Empty function.

sta-wtag replied:
Resolved

bia-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+  },
+  watch:{
+        searchInput: {
+            handler(val)
```
name should be more specific.

sta-wtag replied:
Resolved

bia-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+                },
+                deep: true
+        }
+    },
```
Search functionality is out of scope for this ticket.

sta-wtag replied:
Resolved

hhu-wtag reviewed (changes requested):
Requested some changes here:

1. Prettier isn't working properly as code alignment doesn't seem right.
2. Separate svgs into components.

Other than that, nice work 👍

bia-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,114 @@
+        }
+    },
+     beforeDestroy() {
+     clearTimeout(this.setTime);
```
We rarely use timer to render components.

sta-wtag replied:
Resolved

bia-wtag commented on lang/fr-FR.js:
```diff
@@ -0,0 +1,3 @@
+export default {
+    welcome: 'Bienvenue'
+}
```
Use JSON instead of JS for the translation files.

sta-wtag replied:
Resolved

bia-wtag commented on store/todo/actions.js:
```diff
@@ -0,0 +1,8 @@
+    addFilteredList({state,commit}){
+      commit('setFilteredList',val)
+    }
+  }
```
Please consider the pros and cons of using separate files for various actions, getters, state etc.

sta-wtag replied:
Resolved

bia-wtag commented:
I couldn't run the test locally. Could be an issue with package installation.

Few generic points,

- [ ] Use yarn instead of npm
- [ ] Add the `DS` files to gitignore as they're mac specific
- [ ] Consider separating assets, i.e. svg icons and stylesheets from the component, easier to review and maintain.

sta-wtag committed:
refactor: component created for svgs

sta-wtag committed:
refactor: inline style removed

sta-wtag committed:
refactor:change eslint setup, format code.

sta-wtag committed:
fix: search functionality removed

sta-wtag committed:
refactor: i18n setup without path change

sta-wtag committed:
fix: i18n setup refactor

sta-wtag committed:
fix: lint fix

sta-wtag committed:
i18n: make create button multilingual

sta-wtag committed:
refactor: add mapGetters

hhu-wtag commented on .eslintrc.js:
```diff
@@ -0,0 +1,51 @@
+    'plugin:nuxt/recommended',
+  ],
+  plugins: ['prettier'],
+  rules: {
+    'prettier/prettier': [
+      'warn',
+      {
+        singleQuote: true,
+        semi: true,
+      },
+    ],
+    'padding-line-between-statements': [
+      'error',
+      { blankLine: 'always', prev: ['const', 'let', 'var'], next: '*' },
+      { blankLine: 'always', prev: ['if', 'for', 'import'], next: '*' },
+      { blankLine: 'always', prev: '*', next: ['if', 'for', 'export'] },
+      {
+        blankLine: 'any',
+        prev: ['const', 'let', 'var', 'import', 'export'],
+        next: ['const', 'let', 'var', 'import', 'export'],
+      },
+    ],
+    'newline-before-return': 'error',
+    'vue/no-v-html': 'off',
+    'no-undef': 'off',
+    'no-console': 'off',
+    'no-unused-vars': 'error',
+    'no-useless-catch': 'off',
+    'vue/no-unused-components': [
+      'error',
+      {
+        ignoreWhenBindingPresent: true,
+      },
+    ],
+    quotes: [2, 'single', { avoidEscape: true, allowTemplateLiterals: true }],
+  },
```
We can use a separate config file for prettier. More [here](https://prettier.io/docs/en/configuration.html). Thanks

sta-wtag replied:
Resolved

hhu-wtag commented on components/TaskCard.vue:
```diff
@@ -0,0 +1,38 @@
+  </div>
+</template>
+<script>
+import DeleteButton from './buttons/DeleteButton.vue';
+import EditButton from './buttons/EditButton.vue';
+import TickButton from './buttons/TickButton.vue';
```
We can use relative path here.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,84 @@
+      <div id="searchIcon">
+        <img :src="seachIcon" />
+      </div>
+      <select class="" @change="switchLanguage">
```
We can remove empty class declaration if not needed.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,84 @@
+  </div>
+</template>
+<script>
+import navLogo from '../assets/svg/navLogo.svg';
+import seachIcon from '../assets/svg/searchIcon.svg';
```
ditto

sta-wtag replied:
Resolved

hhu-wtag commented on pages/index.vue:
```diff
@@ -0,0 +1,131 @@
+    this.$nuxt.$on('searchOn', () => {
+      this.taskData = this.getFilteredList;
+    });
+    console.log(this.getTodoList);
```
We can remove the unnecessary console log

sta-wtag replied:
Resolved

hhu-wtag commented on pages/index.vue:
```diff
@@ -0,0 +1,131 @@
+      this.showAddCard = true;
+    },
+    addTask() {
+      console.log(this.$store);
```
ditto

sta-wtag replied:
Resolved

hhu-wtag commented on pages/index.vue:
```diff
@@ -0,0 +1,131 @@
+
+      if (this.task.description.length > 0) {
+        this.task.createdAt = new Date();
+        this.$store.commit('addTask', JSON.parse(JSON.stringify(this.task)));
```
Why are we stringyfying it and then parsing it? Is it because of the `new Date()` ?

sta-wtag replied:
yes

sta-wtag replied:
Resolved

hhu-wtag commented on pages/index.vue:
```diff
@@ -0,0 +1,131 @@
+    this.taskData = this.getTodoList;
+  },
+  methods: {
+    createTask() {
```
```suggestion
    showAddTodoCard() {
```

sta-wtag replied:
Resolved

hhu-wtag commented on store/mutations.js:
```diff
@@ -0,0 +1,9 @@
+export default {
+  addTask(state, val) {
+    console.log(state.states);
```
ditto

sta-wtag replied:
Resolved

hhu-wtag commented on pages/index.vue:
```diff
@@ -0,0 +1,131 @@
+      <div>
+        <button @click="createTask()">{{ $t('create') }}</button>
+      </div>
+      <div>
+        <button class="marginLeft19">All</button>
+        <button class="marginLeft19">Incomplete</button>
+        <button class="marginLeft19">Complete</button>
+      </div>
```
```suggestion
      <div class="filter-buttons">
        <button>All</button>
        <button>Incomplete</button>
        <button>Complete</button>
      </div>
      
      <style lang="scss">
      .filter-buttons > button + button {
         margin-left: 1rem;
      }
      </style>
```

sta-wtag replied:
Resolved

hhu-wtag commented on assets/css/main.css:
```diff
@@ -0,0 +1,83 @@
+    background-color: white;
+}
+
+.spaceBetween {
+    display: flex;
+    justify-content: space-between;
+}
```
`.spaceBetween` should only hold `justify-content: space-between`. 
You can also do the same for flex. Also if you wish to use some atomic classes I would suggest you to make the names more smaller and use kebab casing.

sta-wtag replied:
Resolved

hhu-wtag reviewed (changes requested):
Some changes requested:

- [x] Please remove all the unnecessary console.log()
- [x] Please follow a well documented class naming style such as [BEM](https://getbem.com/)
- [x] For atomic classes, make them super generic and try not to compound classes.

sta-wtag committed:
fix: css class name fix, remove console, set relative path

sta-wtag committed:
fix: lint-fix

sta-wtag committed:
fix: change method name to more task oriented

hhu-wtag commented on components/TaskContainer.vue:
```diff
@@ -0,0 +1,128 @@
+</template>
+<script>
+import { mapGetters } from 'vuex';
+import DeleteButton from '../components/buttons/DeleteButton.vue';
+import noTaskLogo from '../assets/svg/noTask.svg';
```
We can use relative path here.

sta-wtag replied:
Resolved

hhu-wtag commented on components/TaskContainer.vue:
```diff
@@ -0,0 +1,128 @@
+    addTask() {
+      if (this.task.description.length > 0) {
+        this.task.createdAt = new Date().toDateString();
+        // this.$store.commit('addTask', JSON.parse(JSON.stringify(this.task)));
```
We can remove unnecessary comments

hhu-wtag commented on components/TheHeader.vue:
```diff
@@ -0,0 +1,84 @@
+</script>
+<style lang="scss">
+.header {
+  /* width: 100%; */
```
ditto

sta-wtag replied:
Resolved

hhu-wtag reviewed (approved):
Approved with some small change request. Good work 🎉

sta-wtag committed:
fix: lint-fix

bia-wtag reviewed (changes requested):
Few issues that I've noticed,

- [x] Clicking `Add Task` without a todo title results in an error. (attached screenshot)
- [x] Changing locale doesn't translate all the interface, only the create button.
- [x] `You didn’t add any task. Please, add one.` is shown even when there are todos. (attached screenshot)
- [x] Empty todo doesn't create a validation error. 
- [x] Please rename the SVG components from `DeleteButton` to `DeleteIcon` or `DeleteIconButton` as `DeleteButton` is a generic name and doesn't show an SVG.
- [x] CSS classnames should be `kebab-case`, i.e. from `flexGap8` to `flex-gap-8`.
- [x] Further separation of components can be done i.e. creating components for filters, for the placeholder on the homepage, etc.
- [x] IMHO, creating separate getters, actions, and mutations without a folder (todo in this case) will result in a messy store when we'll need more than 1 type of state.

Please look at these issues and refer to Sakib bhai's PR here: https://github.com/sai-wtag/nuxt-todo/pull/1 for pointers. Let me know if anything is not clear. Thanks.

sta-wtag committed:
fix: change icon component names

sta-wtag committed:
Fix:  Resolve issues from PR suggestions

change store folder structure, add multilinguality to all the interfaces, add validation and error message to add task input field

sta-wtag committed:
fix: change classNames to kebab-case

sta-wtag committed:
fix: remove Console

sta-wtag committed:
fix: move list length check to computed

sta-wtag committed:
Update yarn.lock

bia-wtag reviewed (approved):
Tested locally. LGTM.
